distsql: remove queueing in the flow scheduler

yuzefovich opened the pull request:
**flowinfra: de-duplicate cluster flow tests**

Previously, there was a lot of duplication for two cluster flow tests
(one for the single tenant setup and another for the multi tenant
setup), and this commit cleans it up.

Release note: None

**distsql: remove queueing in the flow scheduler**

This commit removes the queueing behavior in the flow scheduler (which
is now renamed to "remote flow runner" to better reflect its purpose).
This behavior was already disabled on 22.2 (with a cluster setting that
could enable it back as an escape hatch) since we wanted to be
conservative when removing it, but I don't foresee any problems with
this, so it should be safe to remove it.

We don't remove the "cancel dead flow coordinator" since it might still
be useful in a mixed-version cluster, but more importantly the
coordinator will be refactored to also cancel the running flows (not
just the queued flows as it it used to). (This change will be in
a separate commit.)

This required removal some of the tests that relied on the queueing
behavior, but I don't think we're losing much test coverage.

Fixes: #34229.

Release note (sql change): `sql.distsql.max_running_flows` cluster
setting has been removed. This setting previously controlled the number
of remote DistSQL flows that a single node would run at any time. Once
that number was exceeded, the incoming flows would get queued until the
number was reduced. This was used as a poor man's version of the
admission control, but now that we have an actual admission control in
place, we don't need that queueing behavior.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/84888)
<!-- Reviewable:end -->

yuzefovich committed:
flowinfra: de-duplicate cluster flow tests

Previously, there was a lot of duplication for two cluster flow tests
(one for the single tenant setup and another for the multi tenant
setup), and this commit cleans it up.

Release note: None

yuzefovich committed:
distsql: remove queueing in the flow scheduler

This commit removes the queueing behavior in the flow scheduler (which
is now renamed to "remote flow runner" to better reflect its purpose).
This behavior was already disabled on 22.2 (with a cluster setting that
could enable it back as an escape hatch) since we wanted to be
conservative when removing it, but I don't foresee any problems with
this, so it should be safe to remove it.

We don't remove the "cancel dead flow coordinator" since it might still
be useful in a mixed-version cluster, but more importantly the
coordinator will be refactored to also cancel the running flows (not
just the queued flows as it it used to). (This change will be in
a separate commit.)

This required removal some of the tests that relied on the queueing
behavior, but I don't think we're losing much test coverage.

Release note (sql change): `sql.distsql.max_running_flows` cluster
setting has been removed. This setting previously controlled the number
of remote DistSQL flows that a single node would run at any time. Once
that number was exceeded, the incoming flows would get queued until the
number was reduced. This was used as a poor man's version of the
admission control, but now that we have an actual admission control in
place, we don't need that queueing behavior.

cucaroach reviewed (approved):
> Reviewed 24 of 24 files at r2, 33 of 33 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/84888)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @michae2)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
